src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -37,10 +34,10 @@ public override void CopyFile(string sourceFullPath, string destFullPath, bool o
             // Copy the contents of the file from the source to the destination, creating the destination in the process
             const int bufferSize = FileStream.DefaultBufferSize;
             const bool useAsync = false;
-            using (Stream src = new FileStream(sourceFullPath, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize, useAsync))
-            using (Stream dst = new FileStream(destFullPath, overwrite ? FileMode.Create : FileMode.CreateNew, FileAccess.ReadWrite, FileShare.None, bufferSize, useAsync))
+            using (var src = new FileStream(sourceFullPath, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize, useAsync))
+            using (var dst = new FileStream(destFullPath, overwrite ? FileMode.Create : FileMode.CreateNew, FileAccess.ReadWrite, FileShare.None, bufferSize, useAsync))
```

Comment:
It's possible. I didn't want to open the handles directly as there's then a bunch of logic to be duplicated, around mapping options, fixing up paths, etc. If we want to try to optimize that, I'd prefer we do it separately. The big win here is in avoiding lots of buffer copies and user/kernel mode transitions.